api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -82,6 +83,24 @@ public static String toSanitizedString(Expression expr) {
     return ExpressionVisitors.visit(expr, new StringSanitizer());
   }
 
+  /**
+   * Extracts an expression that references only the given column IDs from the given expression.
+   *
+   * <p>The result is inclusive. If a row would match the original filter, it must match the result
+   * filter.
+   *
+   * @param expression a filter Expression
+   * @param schema a Schema
+   * @param caseSensitive whether binding is case sensitive
+   * @param ids field IDs used to match predicates to extract from the expression
+   * @return an Expression that selects at least the same rows as the original using only the IDs
+   */
+  public static Expression extractByIdInclusive(
```

Comment:
This has been already merged in #6716.

Reply:
Looks like the tests were missing. I'll add those when I clean this up.